### PR TITLE
funder whitelist first pass

### PR DESCRIPTION
**Pull request by chrisguida**
This feature is not done yet. It adds a whitelist of pubkeys to the `funder` plugin.

### CODE REVIEW

File: plugins/funder.c [hunk 1]
```diff
@@ -43,17 +44,16 @@ static struct pending_open *
 find_channel_pending_open(const struct channel_id *cid)
 {
 	struct pending_open *open;
-	list_for_each(&pending_opens, open, list) {
+	list_for_each(&pending_opens, open, list)
+	{
```
Comment (cdecker) [inline plugins/funder.c:48]: Not compatible with our syntax conventions (newlines only for methods).

File: plugins/funder.c [hunk 2]
```diff
@@ -43,17 +44,16 @@ static struct pending_open *
...
 			return open;
 	}
 	return NULL;
 }
 
 static struct pending_open *
-new_channel_open(const tal_t *ctx,
-		 struct plugin *p,
-		 const struct node_id id,
+new_channel_open(const tal_t *ctx, struct plugin *p, const struct node_id id,
```
Comment (cdecker) [inline plugins/funder.c:56]: Formatting change only?

File: plugins/funder.c [hunk 3]
```diff
@@ -73,17 +73,15 @@ new_channel_open(const tal_t *ctx,
...
 
-static struct command_result *
-unreserve_done(struct command *cmd UNUSED,
-	       const char *buf,
-	       const jsmntok_t *result,
-	       struct pending_open *open)
+static struct command_result *unreserve_done(struct command *cmd UNUSED,
+					     const char *buf,
+					     const jsmntok_t *result,
+					     struct pending_open *open)
```
Comment (cdecker) [inline plugins/funder.c:79]: Formatting change only?

File: plugins/funder.c [hunk 4]
```diff
@@ -73,17 +73,15 @@ new_channel_open(const tal_t *ctx,
...
 {
 	plugin_log(open->p, LOG_DBG,
 		   "`unreserveinputs` for channel %s completed. %*.s",
 		   type_to_string(tmpctx, struct channel_id, &open->channel_id),
-		   json_tok_full_len(result),
-		   json_tok_full(buf, result));
+		   json_tok_full_len(result), json_tok_full(buf, result));
```
Comment (cdecker) [inline plugins/funder.c:84]: Formatting change only?

File: plugins/funder.c [hunk 5]
```diff
@@ -94,15 +92,12 @@ static void unreserve_psbt(struct pending_open *open)
...
-	req = jsonrpc_request_start(open->p, NULL,
-				    "unreserveinputs",
-				    unreserve_done, unreserve_done,
-				    open);
+	plugin_log(
+	    open->p, LOG_DBG, "Calling `unreserveinputs` for channel %s",
+	    type_to_string(tmpctx, struct channel_id, &open->channel_id));
+
+	req = jsonrpc_request_start(open->p, NULL, "unreserveinputs",
+				    unreserve_done, unreserve_done, open);
```
Comment (cdecker) [inline plugins/funder.c:100]: Formatting change only?

**Review by cdecker — CHANGES_REQUESTED** [review]
Thanks @chrisguida for the change. I like it, but the PR has a lot of whitespace and formatting changes, that obscure where the actual changes are. Since this increases chances of conflicts, and overlooking things in the review.

Please remove the formatting changes and resubmit :-)

**Comment by chrisguida** [comment]
> Thanks @chrisguida for the change. I like it, but the PR has a lot of whitespace and formatting changes, that obscure where the actual changes are. Since this increases chances of conflicts, and overlooking things in the review.
> 
> Please remove the formatting changes and resubmit :-)

Yes, apologies for all the formatting changes, I have a default formatter that is very opinionated. I'll be sure to remove them once this is out of draft. Does the CLN team have a formatting settings file for vscode by any chance?

**Comment by niftynei** [comment]
> Does the CLN team have a formatting settings file for vscode by any chance?

Not that I know of, good question for the discord/telegram/irc though.

**Comment by ShahanaFarooqui** [comment]
Closing Draft PRs inactive for more than a year. Please feel free to reopen if required.

**Pull request closed without merging**